### ISSUE TITLE
Does not function with FVTT 7.5

### ISSUE THREAD

**Issue by benpope3** [Foundry-Workshop/ammo-swapper#3]
**Module Version:** v0.1.2

**Describe the bug**
When swapping ammo between two choices, you have to select the current choice to toggle selection to the other choice, and it does not actually switch the ammo being used.

**To Reproduce**
1. Go to 'Ammo Swapper'
2. Left Click on to popup options
3. Select an alternate ammo type 
4. Nothing happens
5. Select the current ammo type
6. The active type displayed switches to the alternate
7. Checking the selection on the weapon or rolling an attack, it does not change the ammo used

**Expected behavior**
Swaps to the selected ammo and the weapon uses the selected ammo.

**Screenshots**

**Using Edge browser**

**Foundry Version 7.5:** 

**ADD5E** 

**Additional context**
DAE, MidiQOL, Furnace, CUB modules enabled

**Comment by davidR1974** [comment]
Have tested a bit around that.. 
When clicking on ammo for a weapon, the ammunition is correctly updated (can be confirmed in character sheet or with Token HUD that shows the correct number of "charges" on the bow)
However ammoSwapper does not renders correctly after the update of the character sheet
If you hit refresh (F5) it shows correct ammo.

Same when changing the amount of ammo in character sheet. ammoSwapper does not reflect automatically but does if you take another action (like swapping ammo)

Maybe a matter of forcing rendering after ammo selection.

**Comment by davidR1974** [comment]
managed to make it work by calling the setAmmunition function in the html on click ammo event listener

```
    html.on('click', '.ammo', (event) => {
      event.stopPropagation();
      const ammo = $(event.currentTarget);
      const weapon = ammo.closest('.weapon');
      const weaponId = weapon.data("weapon-id");
      const ammoId = ammo.data("ammo-id");
      this.manager.setAmmunition(weaponId, ammoId);
      this.manager.setAmmunition(weaponId, ammoId);    /* This line new */
    });
```

**Comment by benpope3** [comment]
Thanks for looking at this davidR1974.  Curious why the "this.manager.setAmmunition(weaponId, ammoId);" added line needs to be repeated twice as shown above.  I can see where it gets inserted in the original code, but how do I add this patch to the code in my world if the module is not updated?

**Comment by davidR1974** [comment]
i modified the AmmoSwapper.js ( ($FOUNDRYDATA)\modules\workshop-ammo-swapper\modules )

as for why you need two calls, i have no explanation and it's probably not the right way correct this (just a workaround with my limited knowledge of the framework). 
The first call get things right (ammo is changed in the item sheet if you open it) but the Ammoswapper does not render to reflect those changes. That second call somewhat forces that.

**Comment by Vilyandil** [comment]
> i modified the AmmoSwapper.js ( ($FOUNDRYDATA)\modules\workshop-ammo-swapper\modules )
> 
> as for why you need two calls, i have no explanation and it's probably not the right way correct this (just a workaround with my limited knowledge of the framework).
> The first call get things right (ammo is changed in the item sheet if you open it) but the Ammoswapper does not render to reflect those changes. That second call somewhat forces that.

I think i have a somewhat explanation to what is happening here.
I used your workaround for a while, then today i tried to understand what happens, and I think is the following:
->click on new ammo triggers ammo swap 
->ammo swap triggers re-rendering (hooked event), render is performed with old ammo because it has not changed yet
Maybe weapon.update() (called in setAmmunition) calls an handler on which the rendering is hooked before the actual swap, but i'm wildly guessing since i don't know the framework.
So I tried to force the rendering to happen after the ammo-update chain by tampering with Init.js, so that when the rendering is requested finds the new ammo: I changed the hooks on "updateActor" and "updateOwnedItem" in this way:

`
Hooks.on("updateActor", (actor, data, options, userId) => {
  if (actor.id === game.user.character?.id) 
	  setTimeout(()=>{ui.ammoSwapper?.render()});
});

Hooks.on("updateOwnedItem", (actor, item, data, options, userId) => {
  if (actor.id === game.user.character?.id) 
	  setTimeout(()=>{ui.ammoSwapper?.render()});
});
`
That setTimeout avoids the immediate render() execution, which is scheduled after the current execution chain.
in the attachment you can find the modified Init.js: [init.zip](https://github.com/Foundry-Workshop/ammo-swapper/files/6095258/init.zip)

**Comment by Forien** [comment]
Fixed in [v0.2.0](https://github.com/Foundry-Workshop/ammo-swapper/releases/tag/v0.2.0)
